pkg/cloudprovider/azure/object_store.go

```diff
@@ -37,10 +41,57 @@ func NewObjectStore(logger logrus.FieldLogger) cloudprovider.ObjectStore {
 	return &objectStore{log: logger}
 }
 
+func getStorageAccountsClient(envVars map[string]string) (*storagemgmt.AccountsClient, error) {
+	spt, err := helpers.NewServicePrincipalTokenFromCredentials(envVars, azure.PublicCloud.ResourceManagerEndpoint)
```

Comment:
We'll need to copy the implementation of `helpers.NewServicePrincipalTokenFromCredentials` and adjust it slightly. It currently panics if it gets an error creating the oauth config.

I would also recommend that we check `envVars` and make sure that `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`, and `AZURE_CLIENT_SECRET` are set before proceeding, or return an error if they're not. Or, perhaps better yet, make `getStorageAccountsClient` take in the 4 required parameters (the 3 previously mentioned, plus the subscription id key) or a struct containing them. Or maybe have it take in `(spt *adal.ServicePrincipalToken, subscriptionIDKey string)` where you get the `spt` before calling it.

Reply:
Makes sense -- we're already using that `helpers` func in the `block_store` so will need to change it there too.